DebugInfomationFormat.md

```diff
@@ -0,0 +1,220 @@
+#Debug Symbol Integration
+
+This is a design proposal for providing debug information within WebAssembly
+that would allow source-level debugging of code compiled into WebAssembly.
+
+## Goals and Principles
+
+The immediate goal is to implement enough debugging functionality in WebAssembly
+MVP to let users perform at least rudimentary debugging on their source.
+Another goal is not to bake in any limitations that would prevent reasonable
+future evolution of the standard.
+
+Specifically, we are currently not focused on identifying a language-independent
+subset of the debug format -- this proposal bundles language-dependent and
+language-independent parts together.
+
+META: We need an evolution strategy that allows new front-end/debugger pairs to
+use the format in the future to transfer information currently unanticipated.
+Examples: a) dynamic scoping in Lisp; b) full DWARF 4 equivalence.
```

Comment:
These examples are just to jog one's imagination for where future evolution may take us.  They're not meant to suggest that we intend to go in that direction, just that we don't want to gratuitously prevent it.  Another example could be "enough of DWARF to allow stock gdb to run on wasm modules".

Anyway, see the discussion below -- we are moving to a different, more flexible design that allows any debug-info format to be used.  I'll mothball this PR soon and create another one with the new design description.